### ISSUE TITLE
Bert pre-training loss convergence

### ISSUE THREAD

**Issue by abhi060698** [google-research/bert#746]
i am trying to pretrain BERT on my own dataset. Even though the code shows that saving and restoring from checkpoints is occuring, my loss value tends to stagnate at around 7. I am not seeing any convergence at all. What can I do?

**Comment by DarrenAbramson** [comment]
Looks like it turned out after all.

**Comment by ekgren** [comment]
> i am trying to pretrain BERT on my own dataset. Even though the code shows that saving and restoring from checkpoints is occuring, my loss value tends to stagnate at around 7. I am not seeing any convergence at all. What can I do?

Did you solve this? If so what was the solution?

**Comment by SongDrink** [comment]
I got the same problem. I try to set my batchsize to a lager number, like 64, It seems to be convergent

**Comment by roberthajdu92** [comment]
I got the same problem too. The problem was, that one of our training dataset (scraped) had thousands of repeated lines, so the training stuck at a local optimum (around 8). Make sure you check for replicas in your dataset before you train.

**Comment by wawltor** [comment]
> I got the same problem too. The problem was, that one of our training dataset (scraped) had thousands of repeated lines, so the training stuck at a local optimum (around 8). Make sure you check for replicas in your dataset before you train.

I got the same problem， could you please tell me which file had the repeated lines？

**Comment by roberthajdu92** [comment]
It's your custom dataset, where you have to look for repeated lines!
